File: atividade5/011.hs
primeiros :: Int -> [t] -> [t]
primeiros _ [] = []
primeiros n (a:b)
    | (n > 0)   = [a] ++ (primeiros (n-1) b)
    | otherwise = []

main = do
    print (primeiros 3 [7, 9, 2, 4, 5, 3])

File: atividades/atividade16/011.pl
primeiros(0, _, []).
primeiros(N, [H|T], [H|TT]) :- NN is N - 1, primeiros(NN, T, TT).

% ?- primeiros(3, [1,2,3,4,5], L).
% L = [1, 2, 3] ;
% false.

% ?- primeiros(0, [1,2,3,4,5], L).
% L = [] ;
% false.

% ?- primeiros(9, [1,2,3,4,5], L).
% false.

% ?- primeiros(1, [], L).
% false.

% ?- primeiros(0, [], L).
% L = [] ;
% false.
